src/layers/trip-layer/trip-utils.js

```diff
@@ -0,0 +1,188 @@
+  const maxCount = 10000;
+  const sampleRawFeatures =
+    allData.length > maxCount ? getSampleData(allData, maxCount, getValue) : allData.map(getValue);
+
+  const features = sampleRawFeatures.map(parseGeoJsonRawFeature);
+
+  let isTrip = false;
+  const featureTypes = getGeojsonFeatureTypes(features);
+
+  // condition 1: contain line string
+  const hasLineString = Boolean(featureTypes.line);
+  if (!hasLineString) {
+    return isTrip;
+  }
+
+  // condition 2:sample line strings contain 4 coordinates
+  const HasLength4 = coordHasLength4(features);
+  if (!HasLength4) {
+    return isTrip;
+  }
+
+  // condition 3:the 4th coordinate of the first feature line strings is valid time
+  const tsHolder = features[0].geometry.coordinates.map(
+    coord => coord[3]
+  );
+
+  const hasValidTime = containValidTime(tsHolder);
+  if (hasValidTime) {
+    isTrip = true;
+  }
+
+  return isTrip;
+}
+
+/**
+ * Get unix timestamp from animatable geojson for deck.gl trip layer
+ * @param {Array<Object>} dataToFeature array of geojson feature objects, can be null
+ * @returns {Array<Number>} unix timestamp in milliseconds
+ */
+export function parseTripGeoJsonTimestamp(dataToFeature) {
+  // Analyze type based on coordinates of the 1st lineString
+  // select a sample trip to analyze time format
+  const empty = {dataToTimeStamp: [], animationDomain: null};
+  const sampleTrip = dataToFeature.find(
+    f =>
+      f &&
+      f.geometry &&
+      f.geometry.coordinates &&
+      f.geometry.coordinates.length >= 3
+  );
+
+  // if no valid geometry
+  if (!sampleTrip) {
+    return empty;
+  }
+
+  const analyzedType = containValidTime(
+    sampleTrip.geometry.coordinates.map(coord => coord[3])
+  );
+
+  if (!analyzedType) {
+    return empty;
+  }
+
+  const {format} = analyzedType;
+  const getTimeValue = coord =>
+    coord && notNullorUndefined(coord[3])
+      ? timeToUnixMilli(coord[3], format)
+      : null;
+
+  const dataToTimeStamp = dataToFeature.map(f =>
+    f && f.geometry && Array.isArray(f.geometry.coordinates)
+      ? f.geometry.coordinates.map(getTimeValue)
+      : null
+  );
+
+  const animationDomain = getAnimationDomainFromTimestamps(dataToTimeStamp);
+
+  return {dataToTimeStamp, animationDomain};
+}
+
+function findMinFromSorted(list = []) {
+  let i = 0;
+  while (i < list.length) {
+    if (notNullorUndefined(list[i])) {
+      return list[i]
+    }
+    i++;
+  }
+  return null;
+}
+
+function findMaxFromSorted(list = []) {
+  let i = list.length - 1;
+  while (i > 0) {
+    if (notNullorUndefined(list[i])) {
+      return list[i]
+    }
+    i--;
+  }
+  return null;
+}
+
+export function getAnimationDomainFromTimestamps(dataToTimeStamp = []) {
+  return dataToTimeStamp.reduce(
+    (accu, tss) => {
+      accu[0] = Math.min(accu[0], findMinFromSorted(tss));
+      accu[1] = Math.max(accu[1], findMaxFromSorted(tss));
+      return accu;
+    },
+    [Number(Infinity), -Infinity]
```

Comment:
I don't think you need to use Number. I tested with the console and:
```
Number(Infinity) === Infinity
true
```

Otherwise, if you want to use Number, it's better to be consistent and use it for both values
```javascript
[Number(Infinity), Number(-Infinity)]
```

Reply:
👍